addons/point_of_sale/models/pos_order.py

```diff
@@ -1412,7 +1412,13 @@ def _prepare_procurement_values(self, group_id=False):
         self.ensure_one()
         # Use the delivery date if there is else use date_order and lead time
         if self.order_id.shipping_date:
-            date_deadline = self.order_id.shipping_date
+            # get timezone from user
+            # and convert to UTC to avoid any timezone issue
+            # because shipping_date is date and date_planned is datetime
+            from_zone = self.env["ir.fields.converter"]._input_tz()
```

Comment:
Maybe use what it used in the pos in general
```suggestion
            from_zone = pytz.timezone(self._context.get('tz') or self.env.user.tz or 'UTC')
```

Reply:
@celm1990 Just keeping consistency with the codebase

Reply:
@adgu-odoo thanks for clarifying. Changes have been made.